src/match/querymatch.c

```diff
@@ -503,12 +503,18 @@ double gt_querymatch_error_rate(GtUword distance,GtUword alignedlen)
 }
 
 bool gt_querymatch_overlap(const GtQuerymatch *querymatch,
-                           GtUword start_relative)
+                           GtUword nextseed_db_end_relative,
+                           GtUword nextseed_query_end_relative,
+                           bool use_db_pos)
 {
   gt_assert(querymatch != NULL);
-  return querymatch->querystart + querymatch->querylen >= start_relative
-           ? true /* overlap with querymatch */
-           : false; /* next seeds starts after curren extension */
+  return querymatch->querystart + querymatch->querylen >
+         nextseed_query_end_relative && (!use_db_pos ||
+         (querymatch->dbstart_relative + querymatch->dblen >
+          nextseed_db_end_relative && nextseed_db_end_relative >=
+          querymatch->dbstart_relative + querymatch->seedlen))
+           ? true   /* overlap with querymatch */
+           : false; /* next seed ends after current extension */
```

Comment:
I absolutely agree with you. I will fix that soon. Bye, Stefan

On 02/13/2016 12:40 , Sascha Steinbiss wrote:
>
> In src/match/querymatch.c 
> <https://github.com/genometools/genometools/pull/738#discussion_r52826855>:
>
> >  {
> >    gt_assert(querymatch != NULL);
> > -  return querymatch->querystart + querymatch->querylen >= start_relative
> > -           ? true /* overlap with querymatch */
> > -           : false; /* next seeds starts after curren extension */
> > +  return querymatch->querystart + querymatch->querylen >
> > +         nextseed_query_end_relative && (!use_db_pos ||
> > +         (querymatch->dbstart_relative + querymatch->dblen >
> > +          nextseed_db_end_relative && nextseed_db_end_relative >=
> > +          querymatch->dbstart_relative + querymatch->seedlen))
> > +           ? true   /* overlap with querymatch */
> > +           : false; /* next seed ends after current extension */
>
> That code is quite convoluted... I think in future cases it might help 
> not trying to do everything in one expression. For the compiler it 
> most likely doesn't matter anyway, but if it was my code I know my 
> future self would thank me later ;)
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/genometools/genometools/pull/738/files#r52826855>.
>


-- 
Prof. Dr. Stefan Kurtz
Zentrum fuer Bioinformatik
Universitaet Hamburg
Bundesstrasse 43
20146 Hamburg
Germany

Email:  kurtz@zbh.uni-hamburg.de
URL:    http://www.zbh.uni-hamburg.de/kurtz
Phone:  +49 (40) 42838 7311
FAX:    +49 (40) 42838 7312